src/lib/common/Makefile

```diff
@@ -53,8 +53,6 @@ obj-core-$(PLATFORM_CONTIKI) += \
 obj-core-$(PLATFORM_LINUX) += \
     sol-platform-linux-common.o \
     sol-board-detect.o
-obj-core-$(PLATFORM_LINUX)-extra-cflags += $(SYSTEMD_CFLAGS)
-obj-core-$(PLATFORM_LINUX)-extra-ldflags += $(SYSTEMD_LDFLAGS)
```

Comment:
just removing it doesn't look right. Maybe add a obj-core-$(PLATFORM_SYSTEMD)... ?

also, we need to link to systemd if either platform-systemd || sd-bus is used.